### PR TITLE
allow wrapped binaries to use a configurable image as the base image

### PR DESCRIPTION
**Pull request by mikedanese**
I'm attempting to dockerize kube-proxy, and failing since it requires iptables. This will allow me to use a custom base image to wrap kube-proxy.

@ihmccreery related to build. Do you mind reviewing?

### CODE REVIEW

File: build/common.sh [hunk 1]
```diff
@@ -737,27 +739,36 @@ function kube::release::sha1() {
 function kube::release::create_docker_images_for_server() {
```
Comment (ihmccreery) [inline build/common.sh]: While you're at it, document the args this function takes.  (See `kube::release::gcs::publish_ci`.)

File: build/common.sh [hunk 2]
```diff
@@ -737,27 +739,36 @@ function kube::release::sha1() {
...
     local binary_name
-    for binary_name in "${KUBE_DOCKER_WRAPPED_BINARIES[@]}"; do
+    for wrappable in "${KUBE_DOCKER_WRAPPED_BINARIES[@]}"; do
+
+      local oldifs=$IFS
+      IFS=","
+      set $wrappable
+      local binary_name="$1"
+      local base_image="$2"
+      IFS=$oldifs
```
Comment (ihmccreery) [inline build/common.sh]: So gross.  Is there no better way to do this?

**Comment by ihmccreery** [comment]
LGTM with nits.

File: build/common.sh (hunk 2 above)
Comment (mikedanese) [inline build/common.sh]: I'm googling a solution, and I can't find a cleaner one then good old save and restore. I could use a regex or expand the entry into an array but the options I'm coming up with are generally grosser. If you have suggestions, I will gladly hear them. Bash is gross :)

**Comment by ihmccreery** [comment]
LGTM.

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Comment by piosz** [comment]
@k8s-bot unit test this please

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Pull request merged**